core/src/test/java/org/kohsuke/stapler/DataBindingTest.java

```diff
@@ -430,4 +432,89 @@ public void testDerivedProperty() {
         DerivedProperty r = bind("{items:[1,3,5]}", DerivedProperty.class);
         assertEquals(Arrays.asList(1, 3, 5), r.getItems());
     }
+
+    @Issue("JENKINS-73690")
+    public void testEnumCanBeNull() throws Exception {
+        MockRequest mr = new MockRequest() {
+            @Override
+            public String getContentType() {
+                return "text/html";
+            }
+        };
+        mr.getParameterMap().put("status", null);
+        RequestImpl req = new RequestImpl(new Stapler(), mr, Collections.emptyList(), null);
+        Object result = new Function.InstanceFunction(
+                        getClass().getMethod("doAcceptEnum", StaplerRequest.class, Status.class))
+                .bindAndInvoke(this, req, null);
+        assertNull(result);
+    }
+
+    public void testEnumIsSupported() throws Exception {
+        MockRequest mr = new MockRequest() {
+            @Override
+            public String getContentType() {
+                return "text/html";
+            }
+        };
+        mr.getParameterMap().put("status", "Done");
+        RequestImpl req = new RequestImpl(new Stapler(), mr, Collections.emptyList(), null);
+        Object result = new Function.InstanceFunction(
+                        getClass().getMethod("doAcceptEnum", StaplerRequest.class, Status.class))
+                .bindAndInvoke(this, req, null);
+        assertEquals(Status.Done, result);
+    }
+
+    public void testEnumUnknownIsThrowing() throws Exception {
+        MockRequest mr = new MockRequest() {
+            @Override
+            public String getContentType() {
+                return "text/html";
+            }
+        };
+        mr.getParameterMap().put("status", "Blabla");
+        RequestImpl req = new RequestImpl(new Stapler(), mr, Collections.emptyList(), null);
+        try {
+            new Function.InstanceFunction(getClass().getMethod("doAcceptEnum", StaplerRequest.class, Status.class))
+                    .bindAndInvoke(this, req, null);
+            fail("Should throw an exception as the enum value is not recognized");
+        } catch (IllegalArgumentException e) {
+            assertEquals(
+                    "No enum constant org.kohsuke.stapler.DataBindingTest.Status.Blabla",
+                    e.getCause().getMessage());
+        }
+    }
+
+    // Just to ensure to keep support for required=true
+    @Issue("JENKINS-73690")
+    public void testEnumCannotBeNullIfRequired() throws Exception {
+        MockRequest mr = new MockRequest() {
+            @Override
+            public String getContentType() {
+                return "text/html";
+            }
+        };
+        mr.getParameterMap().put("status", null);
+        RequestImpl req = new RequestImpl(new Stapler(), mr, Collections.emptyList(), null);
+        try {
+            new Function.InstanceFunction(getClass().getMethod("doRequireEnum", StaplerRequest.class, Status.class))
```

Comment:
Can you use `assertThrows` for cleaner and less code in both these tests?

Reply:
I followed the pattern used in that class file, for the `testMismatchingTypes` test case.

I usually prefer consistent style. Only occurrence of assertThrows is in Stapler2Test, compared to ~20 other occurrences of try/catch.

In a new project or if I was refactoring, yes I would have used assertThrows :)

Reply:
Imo you don't need to add harder to read code because existing code is hard to read.

Reply:
Depend on the perspective. If you look only at the diff, yes I agree, but if you look at the whole file, having different style make it harder. I expect to have more people reading the whole file compared to people reading only this PR/delta.